### PR TITLE
add new flags

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

Add new flags:

astro deployment create/update
- default-task-pod-cpu
- default-task-pod-memory
- resource-quota-cpu
- resource-quota-memory

astro deployment logs
- keyword

## 🎟 Issue(s)

Related #XXX

## 🧪 Functional Testing

manual testing
## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/deployment/deployment.go [hunk 1]
```diff
@@ -153,8 +153,12 @@ func Logs(deploymentID, ws, deploymentName string, warnLogs, errorLogs, infoLogs
 		logLevel = "INFO"
 		i++
 	}
+	if keyword != "" {
+		logLevel = keyword
+		i++
+	}
 	if i > 1 {
-		return errors.New("cannot query for more than one log level at a time")
+		return errors.New("cannot query for more than one log level and or keyword at a time")
```
Comment (kushalmalani) [inline cloud/deployment/deployment.go:161]:
```suggestion
		return errors.New("cannot query for more than one log level and/or keyword at a time")
```

File: cmd/cloud/deployment.go [hunk 2]
```diff
@@ -171,11 +173,14 @@ func newDeploymentCreateCmd(out io.Writer) *cobra.Command {
 			fmt.Println(err)
 		}
 		cmd.Flags().StringVarP(&deploymentType, "type", "", standard, "The Type to use for the Deployment. Possible values can be standard or dedicated.")
+		cmd.Flags().StringVarP(&defaultTaskPodCPU, "default-task-pod-cpu", "", "", "The default task pod CPU to use for the Deployment. Example value: 0.25")
+		cmd.Flags().StringVarP(&defaultTaskPodMemory, "default-task-pod-memory", "", "", "The default taks pod memory to use for the Deployment. Example value: 0.5Gi")
+		cmd.Flags().StringVarP(&resourceQuotaCPU, "resource-quota-cpu", "", "", "The Deployment's CPU resource quota. Example value: 10")
+		cmd.Flags().StringVarP(&resourceQuotaMemory, "resource-quota-memory", "", "", "The Deplyoment's memory resource quota. Example value: 20Gi")
```
Comment (kushalmalani) [inline cmd/cloud/deployment.go:179]:
```suggestion
		cmd.Flags().StringVarP(&resourceQuotaMemory, "resource-quota-memory", "", "", "The Deployment's memory resource quota. Example value: 20Gi")
```

File: cmd/cloud/deployment.go [hunk 3]
```diff
@@ -171,11 +173,14 @@ func newDeploymentCreateCmd(out io.Writer) *cobra.Command {
 			fmt.Println(err)
 		}
 		cmd.Flags().StringVarP(&deploymentType, "type", "", standard, "The Type to use for the Deployment. Possible values can be standard or dedicated.")
+		cmd.Flags().StringVarP(&defaultTaskPodCPU, "default-task-pod-cpu", "", "", "The default task pod CPU to use for the Deployment. Example value: 0.25")
+		cmd.Flags().StringVarP(&defaultTaskPodMemory, "default-task-pod-memory", "", "", "The default taks pod memory to use for the Deployment. Example value: 0.5Gi")
+		cmd.Flags().StringVarP(&resourceQuotaCPU, "resource-quota-cpu", "", "", "The Deployment's CPU resource quota. Example value: 10")
```
Comment (kushalmalani) [inline cmd/cloud/deployment.go:178]: Does the example value needs to have the unit? Also, do we need to provide defaults for these flags?

File: cmd/cloud/deployment.go [hunk 4]
```diff
@@ -171,11 +173,14 @@ func newDeploymentCreateCmd(out io.Writer) *cobra.Command {
 			fmt.Println(err)
 		}
 		cmd.Flags().StringVarP(&deploymentType, "type", "", standard, "The Type to use for the Deployment. Possible values can be standard or dedicated.")
+		cmd.Flags().StringVarP(&defaultTaskPodCPU, "default-task-pod-cpu", "", "", "The default task pod CPU to use for the Deployment. Example value: 0.25")
+		cmd.Flags().StringVarP(&defaultTaskPodMemory, "default-task-pod-memory", "", "", "The default taks pod memory to use for the Deployment. Example value: 0.5Gi")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:177]:
```suggestion
		cmd.Flags().StringVarP(&defaultTaskPodMemory, "default-task-pod-memory", "", "", "The default task pod memory to use for the Deployment. Example value: 0.5Gi")
```

File: cloud/deployment/deployment_test.go [hunk 5]
```diff
@@ -813,17 +813,17 @@ func TestLogs(t *testing.T) {
...
 		mockPlatformCoreClient.AssertExpectations(t)
 		mockCoreClient.AssertExpectations(t)
 	})
 
 	t.Run("query for more than one log level error", func(t *testing.T) {
-		err := Logs(deploymentID, ws, "", true, true, true, logCount, mockPlatformCoreClient, mockCoreClient)
+		err := Logs(deploymentID, ws, "", "", true, true, true, logCount, mockPlatformCoreClient, mockCoreClient)
 		assert.Error(t, err)
-		assert.Contains(t, err.Error(), "cannot query for more than one log level at a time")
+		assert.Equal(t, err.Error(), "cannot query for more than one log level and or keyword at a time")
```
Comment (lzdanski) [inline cloud/deployment/deployment_test.go:826]:
```suggestion
		assert.Equal(t, err.Error(), "cannot query for more than one log level and/or keyword at a time")
```
To keep consistent if accepting the suggestion from other review comment.

File: cmd/cloud/deployment.go [hunk 6]
```diff
@@ -214,6 +217,13 @@ func newDeploymentUpdateCmd(out io.Writer) *cobra.Command {
 	if organization.IsOrgHosted() {
 		cmd.Flags().StringVarP(&schedulerSize, "scheduler-size", "", "", "The size of Scheduler for the Deployment. Possible values can be small, medium, large")
 		cmd.Flags().StringVarP(&highAvailability, "high-availability", "a", "", "Enables High Availability for the Deployment")
+		cmd.Flags().StringVarP(&defaultTaskPodCPU, "default-task-pod-cpu", "", "", "The Default Task Pod CPU to use for the Deployment. Example value: 0.25")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:220]: Similar to other comment - this number doesn't include units and might need them if we add them to the other one.

File: cmd/cloud/deployment.go [hunk 7]
```diff
@@ -214,6 +217,13 @@ func newDeploymentUpdateCmd(out io.Writer) *cobra.Command {
...
+		cmd.Flags().StringVarP(&defaultTaskPodMemory, "default-task-pod-memory", "", "", "The Default Taks Pod Memory to use for the Deployment. Example value: 0.5Gi")
+		cmd.Flags().StringVarP(&resourceQuotaCPU, "resource-quota-cpu", "", "", "The Resource Quota CPU to use for the Deployment. Example value: 10")
+		cmd.Flags().StringVarP(&resourceQuotaMemory, "resource-quota-memory", "", "", "The Resource Quota Memory to use for the Deployment. Example value: 20Gi")
+	} else {
+		cmd.Flags().IntVarP(&updateSchedulerAU, "scheduler-au", "s", 0, "The Deployment's Scheduler resources in AUs")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:225]:
```suggestion
		cmd.Flags().IntVarP(&updateSchedulerAU, "scheduler-au", "s", 0, "The Deployment's Scheduler resources in AUs.")
```

File: cmd/cloud/deployment.go [hunk 8]
```diff
@@ -214,6 +217,13 @@ func newDeploymentUpdateCmd(out io.Writer) *cobra.Command {
...
+		cmd.Flags().IntVarP(&updateSchedulerReplicas, "scheduler-replicas", "r", 0, "The number of Scheduler replicas for the Deployment")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:226]:
```suggestion
		cmd.Flags().IntVarP(&updateSchedulerReplicas, "scheduler-replicas", "r", 0, "The number of Scheduler replicas for the Deployment.")
```

File: cmd/cloud/deployment.go [hunk 9]
```diff
@@ -134,6 +135,7 @@ func newDeploymentLogsCmd() *cobra.Command {
 	cmd.Flags().BoolVarP(&warnLogs, "warn", "w", false, "Show logs with a log level of 'warning'")
 	cmd.Flags().BoolVarP(&errorLogs, "error", "e", false, "Show logs with a log level of 'error'")
 	cmd.Flags().BoolVarP(&infoLogs, "info", "i", false, "Show logs with a log level of 'info'")
+	cmd.Flags().StringVarP(&logsKeyword, "keyword", "k", "", "Show logs that contains this keyword or phrase")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:138]:
```suggestion
	cmd.Flags().StringVarP(&logsKeyword, "keyword", "k", "", "Show logs that contains this keyword or phrase.")
```
Can the search use fuzzy matching or does it require a string match?

**Review by lzdanski — COMMENTED** [review]
Some things for consistency - small fixes!

File: cmd/cloud/deployment.go (hunk 3 above)
Comment (sunkickr) [reply cmd/cloud/deployment.go:178]: The defaults come from the CORE API if "" is passed. For some reason cpu does not have units and memory does. Just following what is passed to the CORE API. Got these examples from the CORE API docs and they work

File: cmd/cloud/deployment.go (hunk 6 above)
Comment (sunkickr) [reply cmd/cloud/deployment.go:220]: CORE API does not want units for CPU

File: cmd/cloud/deployment.go (hunk 9 above)
Comment (sunkickr) [reply cmd/cloud/deployment.go:138]: I am not sure. I believe it is an extra string

**Comment by sunkickr** [comment]
@kushalmalani  committed requested changes

**Review by kushalmalani — APPROVED** [review]

File: cmd/cloud/deployment.go [hunk 10]
```diff
@@ -134,6 +135,7 @@ func newDeploymentLogsCmd() *cobra.Command {
 	cmd.Flags().BoolVarP(&warnLogs, "warn", "w", false, "Show logs with a log level of 'warning'")
 	cmd.Flags().BoolVarP(&errorLogs, "error", "e", false, "Show logs with a log level of 'error'")
 	cmd.Flags().BoolVarP(&infoLogs, "info", "i", false, "Show logs with a log level of 'info'")
+	cmd.Flags().StringVarP(&logsKeyword, "keyword", "k", "", "Show logs that contains this keyword or phrase.")
```
Comment (lzdanski) [inline cmd/cloud/deployment.go:138]:
```suggestion
	cmd.Flags().StringVarP(&logsKeyword, "keyword", "k", "", "Show logs that contain this exact keyword or phrase.")
```

If no fuzzy searching, probably need to imply that it needs to be an exact match + typo

**Review by lzdanski — APPROVED** [review]
Small suggestion + typo fix!

**Review by vandyliu — APPROVED** [review]

**Pull request merged**